assets/app/scripts/services/data.js

```diff
@@ -270,24 +272,45 @@ angular.module('openshiftConsole')
         return;
       }
 
-      self.create(resource, null, object, context, opts).then(
-        function (data) {
-          // include the original object, so the error handler can display the kind/name
-          data.object = object;
-          successResults.push(data);
-          remaining--;
-          _checkDone();
-        },
-        function (data) {
-          // include the original object, so the handler can display the kind/name
-          data.object = object;
-          failureResults.push(data);
-          remaining--;
-          _checkDone();
-        }
-      );
+      var success = function(data) {
+        // include the original object, so the error handler can display the kind/name
+        data.object = object;
+        successResults.push(data);
+        remaining--;
+        _checkDone();
+      };
+      var failure = function(data) {
+        // include the original object, so the handler can display the kind/name
+        data.object = object;
+        failureResults.push(data);
+        remaining--;
+        _checkDone();
+      };
+
+      switch(action) {
+      case "create":
+        self.create(resource, null, object, context, opts).then(success, failure);
+        break;
+      case "update":
+        self.update(resource, object.metadata.name, object, context, opts).then(success, failure);
+        break;
+      default:
+        // default case to prevent unspecified actions and typos
+        failureResults.push({
+          data: {
+            message: "Invalid '" + action + "'  action."
+          },
+          status: 400,
+          headers: function() { return null; },
+          config: {},
+          object: object
+        });
```

Comment:
I don't think we should report errors this way. Does it work if you just call `deferred.reject`? You could use `_.forEach` instead of `Array.forEach` and return `false` to break out of the loop early when it happens.